File: app/src/main/java/com/scrap2stack/app/data/remote/dto/AuthDto.kt
package com.scrap2stack.app.data.remote.dto

data class RegisterRequest(
    val name: String,
    val username: String,
    val email: String,
    val password: String
)

data class LoginRequest(
    val email: String,
    val password: String
)

data class AuthResponse(
    val token: String,
    val user: UserSummaryDto
)

data class UserSummary(
    val id: String,
    val name: String,
    val username: String,
    val email: String
)

// Alias for consistency with backend DTO naming
typealias UserSummaryDto = UserSummary

File: backend/internal/dto/auth_dto.go
package dto

type RegisterRequest struct {
	Name     string `json:"name" binding:"required"`
	Username string `json:"username" binding:"required"`
	Email    string `json:"email" binding:"required,email"`
	Password string `json:"password" binding:"required,min=6"`
}

type LoginRequest struct {
	Email    string `json:"email" binding:"required,email"`
	Password string `json:"password" binding:"required"`
}

type AuthResponseData struct {
	Token string      `json:"token"`
	User  UserSummary `json:"user"`
}

type UserSummary struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	Username string `json:"username"`
	Email    string `json:"email"`
}
